test/test_docker_api_deploy.py

```diff
@@ -43,129 +74,140 @@ def test_docker_api_deploy(self):
+        else:
+            result = deploy_func()
+        return result
 
-        self.assertEqual(result_alter_session_cmd, expected_alter_session_cmd)
-
-        self.assertIn(
-            f".build_output/cache/exports/test-flavor-release-",
-            deploy_result.release_path,
-        )
-
-        self.assertEqual(
-            deploy_result.human_readable_upload_location,
-            f"http://{self.docker_environment.database_host}:{self.docker_environment.ports.bucketfs}/"
-            f"{bucket_name}/{path_in_bucket}/test-flavor-release-{release_name}.tar.gz",
-        )
-
-        expected_path_in_bucket = (
-            bfs.path.build_path(
-                backend=bfs.path.StorageBackend.onprem,
-                url=f"http://{self.docker_environment.database_host}:{self.docker_environment.ports.bucketfs}",
-                bucket_name=bucket_name,
-                service_name=bucketfs_name,
-                username="w",
-                password=self.docker_environment.bucketfs_password,
-                verify=False,
-                path=path_in_bucket,
-            )
-            / f"test-flavor-release-{release_name}.tar.gz"
-        )
-
-        # Compare UDF path of `bucket_path` until bfs.path.PathLike implements comparison
-        self.assertEqual(
-            expected_path_in_bucket.as_udf_path(),
-            deploy_result.bucket_path.as_udf_path(),
-        )
-
-        self.validate_file_on_bucket_fs(
-            bucket_name, f"{path_in_bucket}/test-flavor-release-{release_name}.tar.gz"
-        )
-
-    def test_docker_api_deploy_without_path_in_bucket(self):
+    def _validate_deploy(
+        self, compression: bool, path: Optional[str], expected_extension: str
+    ):
         release_name = "TEST"
         bucketfs_name = "bfsdefault"
         bucket_name = "default"
         flavor_path = exaslct_utils.get_test_flavor()
-        result = api.deploy(
-            flavor_path=(str(flavor_path),),
-            bucketfs_host=self.docker_environment.database_host,
-            bucketfs_port=self.docker_environment.ports.bucketfs,
-            bucketfs_user=self.docker_environment.bucketfs_username,
-            bucketfs_password=self.docker_environment.bucketfs_password,
-            target_docker_repository_name=self.test_environment.docker_repository_name,
-            bucketfs_use_https=False,
-            bucketfs_name=bucketfs_name,
-            bucket=bucket_name,
-            release_name=release_name,
+        result = self._run_deploy(
+            bucket_name, bucketfs_name, compression, flavor_path, path, release_name
         )
+
         self.assertIn(str(flavor_path), result.keys())
         self.assertEqual(len(result), 1)
         self.assertIn(str(flavor_path), result.keys())
         self.assertEqual(len(result[str(flavor_path)]), 1)
         deploy_result = result[str(flavor_path)]["release"]
+
+        complete_path_in_bucket = "/".join(
+            filter(
+                None,
```

Comment:
I know that None is documented in the filter function as the identity, but I think, I would prefer a more obvious comparison, because if I need to look at the documentation of a function to find out what weird behavior it might have, it is not easy to understand. Alternative, we could also use list comprehension